.github/workflows/e2e.yml

```diff
@@ -0,0 +1,107 @@
+name: End-to-end Test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  without-auditor:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.CR_PAT }}
+
+    - name: Start up a Local Scalar DL Network
+      working-directory: .github
+      run: |
+        git init
+        git pull https://${{ github.repository_owner }}:${{ secrets.CR_PAT }}@github.com/scalar-labs/scalardl-samples.git 3.3
+        docker-compose -f docker-compose.yml -f expose-cassandra-port.yml up -d cassandra
+        echo -n "Wait for C* "
+        until docker exec scalardl-samples-cassandra-1 cqlsh 2> /dev/null ; do
+          sleep 0.5
+          echo -n "#"
+        done
+        echo ""
+        docker cp function_schema.cql scalardl-samples-cassandra-1:function_schema.cql
+        docker exec scalardl-samples-cassandra-1 cqlsh -f function_schema.cql
+        docker-compose -f docker-compose.yml -f expose-cassandra-port.yml up scalardl-ledger-schema-loader-cassandra
+        docker-compose -f docker-compose.yml -f expose-cassandra-port.yml up -d scalar-ledger ledger-envoy
+        sleep 5
+
+    - name: NPM Install
```

Comment:
Maybe can the node modules package cache feature to speed up the `npm install` process?

```
- name: Cache dependencies
        uses: actions/cache@v2
        with:
          path: ~/.npm
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-node-
```

Reply:
Thank you, Yves.
I found that maybe I can leverage the `setup-node` action's built-in caching feature.
https://github.com/actions/setup-node

I will do that.